plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/DirectoryLister.java

```diff
@@ -27,4 +27,7 @@
 {
     RemoteIterator<LocatedFileStatus> list(FileSystem fs, Table table, Path path)
             throws IOException;
+
+    RemoteIterator<LocatedFileStatus> listFilesRecursively(FileSystem fs, Table table, Path path)
```

Comment:
That’s incorrect, this method will list all contents at the requested path (shallowly) including directories, which is why the HiveFileIterator previously handled recursing through child directories (or failing, or ignoring them as per whatever nested directory policy was set).

Reply:
I fail to see a constellation in which directories would be returned by the `list` method with the current `NestedDirectoryPolicy` values: `IGNORED`, `RECURSE`, `FAIL`.

It would probably make sense to put a separate (preparatory) PR containing the newly added hive test to make sure that there is no regression performed through your latest changes. Just to be on the safe-side.

Reply:
Got it. I was fixed previously on the `HiveFileIterator`, but noticed now that the discussion is about the `DirectoryLister` and its concrete implementation.